homeassistant/components/binary_sensor/volvooncall.py

```diff
@@ -25,14 +26,9 @@ class VolvoSensor(VolvoEntity, BinarySensorDevice):
     @property
     def is_on(self):
         """Return True if the binary sensor is on."""
-        val = getattr(self.vehicle, self._attribute)
-        if self._attribute == 'bulb_failures':
-            return bool(val)
-        if self._attribute in ['doors', 'windows']:
-            return any([val[key] for key in val if 'Open' in key])
-        return val != 'Normal'
+        return self.instrument.is_on
 
     @property
     def device_class(self):
         """Return the class of this sensor, from DEVICE_CLASSES."""
-        return 'safety'
+        return self.instrument.device_class
```

Comment:
Validate that the device class of the instrument is in the constant list of valid device classes from the binary sensor platform, before returning it here.

Reply:
✓